src/fiatconnect/saga.ts

```diff
@@ -711,6 +715,60 @@ export function* handleSelectFiatConnectQuote({
   }
 }
 
+export function* handlePostKyc({ payload }: ReturnType<typeof postKycAction>) {
+  const { quote } = payload
+  const kycSchema = quote.getKycSchema()
+
+  try {
+    if (!kycSchema) {
+      // this should never happen
+      throw new Error('Unexpected error, quote is missing kyc schema')
+    }
+    yield call(postKyc, {
+      providerInfo: quote.getProviderInfo(),
+      kycSchema,
+    })
+    // We also need to save a user's quote parameters so we can re-fetch if KYC takes a long
+    // time to process.
+    yield put(
+      cacheQuoteParams({
+        providerId: quote.getProviderId(),
+        kycSchema,
+        cachedQuoteParams: {
+          cryptoAmount: quote.getCryptoAmount(),
+          fiatAmount: quote.getFiatAmount(),
+          flow: quote.flow,
+          cryptoType: quote.getCryptoType(),
+          fiatType: quote.getFiatType(),
+        },
+      })
+    )
```

Comment:
These two lines are duplicated in `handleFiatConnectQuote` and does exactly the same for the case when persona is approved, but kyc is missing on provider. I didn't think it was worth the effort to share the code, since both needs to fire a different "completion" action (personaFinished vs selectFiatConnectQuoteCompleted) at the end